pandas/tests/generic/test_label_or_level_utils.py

```diff
@@ -0,0 +1,368 @@
+import pytest
+import pandas as pd
+import pandas.util.testing as tm
+from pandas.core.dtypes.missing import array_equivalent
+
+
+# Fixtures
+# ========
+@pytest.fixture
+def df():
+    """DataFrame with columns 'L1', 'L2', and 'L3' """
+    return pd.DataFrame({'L1': [1, 2, 3],
+                         'L2': [11, 12, 13],
+                         'L3': ['A', 'B', 'C']})
+
+
+@pytest.fixture(params=[[], ['L1'], ['L1', 'L2'], ['L1', 'L2', 'L3']])
+def df_levels(request, df):
+    """DataFrame with columns or index levels 'L1', 'L2', and 'L3' """
+    levels = request.param
+
+    if levels:
+        df = df.set_index(levels)
+
+    return df
+
+
+@pytest.fixture
+def df_ambig(df):
+    """DataFrame with levels 'L1' and 'L2' and labels 'L1' and 'L3' """
+    df = df.set_index(['L1', 'L2'])
+
+    df['L1'] = df['L3']
+
+    return df
+
+
+@pytest.fixture
+def df_duplabels(df):
+    """DataFrame with level 'L1' and labels 'L2', 'L3', and 'L2' """
+    df = df.set_index(['L1'])
+    df = pd.concat([df, df['L2']], axis=1)
+
+    return df
+
+
+# Test is label/level reference
+# =============================
+def get_labels_levels(df_levels):
+    expected_labels = list(df_levels.columns)
+    expected_levels = [name for name in df_levels.index.names
+                       if name is not None]
+    return expected_labels, expected_levels
+
+
+def check_label_reference(frame, labels, axis):
```

Comment:
preface these by assert_ in the function name (check is ok, but assert_ is more inline with the codebase)

Reply:
renamed